Jenkinsfile

```diff
@@ -0,0 +1,289 @@
+       OS_PROJECT_DOMAIN_NAME = "Default"
+       OS_USER_DOMAIN_NAME = "Default"
+       OS_PROJECT_NAME = "orchestrator"
+       OS_TENANT_NAME = "orchestrator"
+       OS_USERNAME = credentials('os-username')
+       OS_PASSWORD = credentials('os-password')
+       OS_AUTH_URL = credentials('os-auth-url')
+       OS_INTERFACE = "public"
+       OS_IDENTITY_API_VERSION = "3"
+       OS_REGION_NAME = "RegionOne"
+       OS_AUTH_PLUGIN = "password"
+       // ANSIBLE SETTINGS
+       ANSIBLE_TIMEOUT = "60"
+
+       // ROOT X.509 CERTIFICATES
+       ca_crt_file = credentials('xopera-ca-crt')
+       ca_key_file = credentials('xopera-ca-key')
+
+       // CI-CD vars
+       // When triggered from git tag, $BRANCH_NAME is actually GIT's tag_name
+       TAG_SEM_VER_COMPLIANT = """${sh(
+                returnStdout: true,
+                script: './validate_tag.sh SemVar $BRANCH_NAME'
+            )}"""
+
+       TAG_MAJOR_RELEASE = """${sh(
+                returnStdout: true,
+                script: './validate_tag.sh MajRel $BRANCH_NAME'
+            )}"""
+
+       TAG_PRODUCTION = """${sh(
+                returnStdout: true,
+                script: './validate_tag.sh production $BRANCH_NAME'
+            )}"""
+
+       TAG_STAGING = """${sh(
+                returnStdout: true,
+                script: './validate_tag.sh staging $BRANCH_NAME'
+            )}"""
+   }
+    stages {
+        stage ('Pull repo code from github') {
+            steps {
+                checkout scm
+            }
+        }
+        stage('Inspect GIT TAG'){
+            steps {
+                sh """ #!/bin/bash
+                echo 'TAG: $BRANCH_NAME'
+                echo 'Tag is compliant with SemVar 2.0.0: $TAG_SEM_VER_COMPLIANT'
+                echo 'Tag is Major release: $TAG_MAJOR_RELEASE'
+                echo 'Tag is production: $TAG_PRODUCTION'
+                echo 'Tag is staging: $TAG_STAGING'
+                """
+            }
+
+        }
+        stage('Test MODAK') {
+            steps {
+                sh  """ #!/bin/bash -l
+                set -x
+                set -eux
+                cd MODAK/
+
+                python3 -m venv venv-pre-commit
+                . venv-pre-commit/bin/activate
+                python3 -m pip install --upgrade pip
+                python3 -m pip install --no-cache-dir pre-commit
+                pre-commit run -a
+                """
+                sh '''#! /bin/bash
+                set -x
+                cd MODAK/
+
+                docker-compose down -v || :
+                if [ -n "\$(docker ps | grep modak)" ]; then
+                    docker kill \$(docker ps | grep modak | awk '{print \$1}') || :
+                fi
+
+                docker-compose -f docker-compose.yml.jenkins up -V --build --force-recreate --always-recreate-deps -d
+                docker cp db/modak_mysqldump.sql \$(docker ps | grep modak | grep sql | awk '{print \$1}'):/docker-entrypoint-initdb.d/
```

Comment:
This seems like a race condition. Is the bind mount not working?

Reply:
In case this failed on an SELinux-enabled host (RHEL/CentOS), the `:z` option for the bind mount should ensure the relabeling.

Reply:
Yes, it's a race condition. I tried the "z" option and it still did not work. I'm content with this solution because we only need it until we no longer need a database to run the unit tests -- at which point we can stop spinning up docker instances and just run the unit tests. Since I think that's coming soon, I think we won't be at risk from this for long.